runtime/runtime-common/src/precompiles.rs

```diff
@@ -120,8 +122,8 @@ impl<R> GoldenGatePrecompiles<R> {
 	/// * 0x5002 - is Sr25519 verify
 	/// * 0x5003 - is Ecdsa verify
 	/// * 0x5005 - is cross virtual machine (XVM)
-	pub fn used_addresses() -> [H160; 17] {
```

Comment:
isn't impl produce a virtual table that may reduce performance, I would keep array.

Reply:
update from https://github.com/AstarNetwork/Astar/blob/feat/polkadot-v0.9.43-uplift/runtime/astar/src/precompiles.rs#L58, should be easy upgrade in the future

 pub fn used_addresses() -> impl Iterator<Item = H160> {
        sp_std::vec![1, 2, 3, 4, 5, 6, 7, 8, 1024, 1025, 1026, 1027, 20481, 20482, 20483, 20484]
            .into_iter()
            .map(hash)
    }

Reply:
Yeah, but it doesn't particularly matter in such cases. But it's fine, I guess.